_posts/2020-07-21-cube-functionality-to-kartothek.markdown

```diff
@@ -0,0 +1,409 @@
+...     return {
+...         "transformed": df.loc[
+...             :,
+...             [
+...                 "avg_temp_country_min",
+...                 "city",
+...                 "country",
+...                 "day",
+...             ]
+...         ],
+...     }
+>>> transformed = query_cube_bag(
+...     cube=cube,
+...     store=store_factory,
+...     partition_by="day",
+... ).map(transform)
+>>> datasets_transformed = extend_cube_from_bag(
+...     data=transformed,
+...     store=store_factory,
+...     cube=cube,
+...     ktk_cube_dataset_ids=["transformed"],
+... ).compute()
+>>> query_cube(
+...     cube=cube,
+...     store=store,
+...     payload_columns=[
+...         "avg_temp",
+...         "avg_temp_country_min",
+...     ],
+... )[0]
+   avg_temp  avg_temp_country_min     city country        day
+0         8                     6  Dresden      DE 2020-07-01
+1         4                     4  Dresden      DE 2020-07-02
+2         6                     6  Hamburg      DE 2020-07-01
+3         5                     4  Hamburg      DE 2020-07-02
+4         6                     6   London      UK 2020-07-01
+5         8                     4   London      UK 2020-07-02
+```
+Notice that the `partition_by` argument does not have to match the cube `partition_columns` to work. 
+You may use any indexed column. Keep in mind that fine-grained partitioning can have drawbacks though, 
+namely large scheduling overhead and many blob files which can make reading the data inefficient.
+
+```python
+>>> print_filetree(store, "geodata++transformed")
+geodata++transformed.by-dataset-metadata.json
+geodata++transformed/table/_common_metadata
+geodata++transformed/table/country=DE/<uuid>.parquet
+geodata++transformed/table/country=DE/<uuid>.parquet
+geodata++transformed/table/country=UK/<uuid>.parquet
+geodata++transformed/table/country=UK/<uuid>.parquet
+```
+
+## Append
+New rows can be added to the cube using an append operation.
+
+```python
+>>> from kartothek.io.eager_cube import append_to_cube
+>>> df_weather2 = pd.read_csv(
+...     filepath_or_buffer=StringIO("""
+... avg_temp     city country        day
+...       20 Santiago      CL 2020-07-01
+...       22 Santiago      CL 2020-07-02
+...     """.strip()),
+...     delim_whitespace=True,
+...     parse_dates=["day"],
+... )
+>>> datasets_appended = append_to_cube(
+...   data=df_weather2,
+...   store=store,
+...   cube=cube,
+... )
+>>> print_filetree(store, "geodata++seed")
+geodata++seed.by-dataset-metadata.json
+geodata++seed/indices/city/<ts>.by-dataset-index.parquet
+geodata++seed/indices/city/<ts>.by-dataset-index.parquet
+geodata++seed/indices/day/<ts>.by-dataset-index.parquet
+geodata++seed/indices/day/<ts>.by-dataset-index.parquet
+geodata++seed/table/_common_metadata
+geodata++seed/table/country=CL/<uuid>.parquet
+geodata++seed/table/country=DE/<uuid>.parquet
+geodata++seed/table/country=UK/<uuid>.parquet
+```
+Notice that the indices where updated automatically.
```

Comment:
```suggestion
Indices are updated automatically. If we query the data now, we can see that only the seed dataset got updated but the additional columns are missing:
```

The example below does not describe anything index-specific but is a great example for the left-join semantic